File: 16-express-js-hexa/adapters/persistence/inMemoryUserRepository.js
import { UserRepository } from '../../application/userRepository.js'
import { UserDPO } from './dpos/userDPO.js';

export class InMemoryUserRepository extends UserRepository {
    constructor() {
        super();
        this.users = new Map();
    }

    saveUser(user) {
        // Explizite Umwandlung in DPO
        const userDPO = UserDPO.fromDomain(user);
        this.users.set(`${userDPO.id}`, userDPO);
        console.log(this.users, userDPO.id, userDPO, user.id);
        return Promise.resolve(UserDPO.toDomain(userDPO));
    }

    findUserById(id) {
        const userDPO = this.users.get(`${id}`);
        console.log(this.users, id);
        if (!userDPO) return Promise.resolve(null);
        // Explizite Umwandlung zurück in Domain-Modell
        return Promise.resolve(UserDPO.toDomain(userDPO));
    }
}

File: 16b-fastapi-python-hexa/adapters/persistence/in_memory_user_repository.py
from typing import Optional, Dict
from application.user_repository import UserRepository
from domain.models.user import User
from .dpos.user_dpo import UserDPO

class InMemoryUserRepository(UserRepository):
    def __init__(self):
        self.users: Dict[str, UserDPO] = {}

    async def save_user(self, user: User) -> User:
        user_dpo = UserDPO.from_domain(user)
        self.users[user_dpo.id] = user_dpo
        return user_dpo.to_domain()

    async def find_user_by_id(self, id: str) -> Optional[User]:
        user_dpo = self.users.get(id)
        return user_dpo.to_domain() if user_dpo else None